### PR TITLE
Update az aro extension to support specifying --pull-secret file without @ prefix

### PR DESCRIPTION
**Pull request by s-fairchild**
Update az aro extension to handle reading the pull-secret file directly without having to prefix the path with @.
The pull-secret file can still be specified with or without @ as  `--pull-secret @pull-secret.txt`, or `--pull-secret pull-secret.txt`.

### Which issue this PR addresses:
https://github.com/Azure/ARO-RP/issues/2110

Fixes

### What this PR does / why we need it:
The pull secret file has to be prefixed with the @ symbol rather than the file path directly. Also brought up here: https://github.com/MicrosoftDocs/azure-docs/issues/53604

### Test plan for issue:
Tested locally with `az aro create -g $RESOURCEGROUP --name steven-cluster-2 --vnet aro-vnet --master-subnet master-subnet --worker-subnet worker-subnet --pull-secret ~/Downloads/pull-secret`

Unit tests pass, filesystem mocking will need to be done to test reading a file in addition to json strings.

### Is there any documentation that needs to be updated for this PR?
Azure docs may want to be updated, although it will work with the existing documentation.

### CODE REVIEW

**Review by cadenmarchese — APPROVED** [review]

**Review by s-amann — APPROVED** [review]
lgtm, nice work for a quick win

**Review by SudoBrendan — APPROVED** [review]
This LGTM, but if we are changing how customers interact with the CLI without backwards compatibility (unclear to me if this is happening - do both now work - with or without `@`?), we might need to talk to Jerome about getting a customer notification sent out before merging this upstream.

**Comment by s-fairchild** [comment]
@SudoBrendan JSON can be passed as a string or file with this change, so using `--pull-secret @pull_secret.txt` works the same as `--pull-secret pull_secret.txt`.
We may want to ask them to update their documentation to show that `@` isn't required when referencing a file.

**Pull request merged**
